skate-plugin/project-gen/src/jvmMain/kotlin/slack/tooling/aibot/ChatWindowUi.kt

```diff
@@ -0,0 +1,178 @@
+import androidx.compose.ui.text.input.TextFieldValue
+import androidx.compose.ui.unit.dp
+import org.jetbrains.jewel.foundation.theme.JewelTheme
+import org.jetbrains.jewel.ui.component.Icon
+import org.jetbrains.jewel.ui.component.IconButton
+import org.jetbrains.jewel.ui.component.Text
+import org.jetbrains.jewel.ui.component.TextArea
+
+object ChatWindowUi {
+  @Composable
+  fun ChatWindow(state: ChatScreen.State, modifier: Modifier = Modifier) {
+    Column(modifier = modifier.fillMaxSize().background(JewelTheme.globalColors.paneBackground)) {
+      LazyColumn(modifier = Modifier.weight(1f), reverseLayout = true) {
+        items(state.messages.reversed()) { message ->
+          Row(
+            modifier = Modifier.fillMaxWidth(),
+            horizontalArrangement = if (message.isMe) Arrangement.End else Arrangement.Start,
+          ) {
+            ChatBubble(message)
+          }
+        }
+      }
+      ConversationField(
+        modifier = Modifier,
+        onSendMessage = { userMessage ->
+          state.eventSink(ChatScreen.Event.SendMessage(userMessage))
+        },
+      )
+    }
+  }
+
+  @Composable
+  fun ConversationField(modifier: Modifier = Modifier, onSendMessage: (String) -> Unit) {
+    var textValue by remember { mutableStateOf(TextFieldValue()) }
+    val isTextNotEmpty = textValue.text.isNotBlank()
+
+    fun sendMessage() {
+      if (isTextNotEmpty) {
+        onSendMessage(textValue.text)
+        textValue = TextFieldValue("")
+      }
+    }
+
+    Row(
+      modifier.padding(4.dp).height(IntrinsicSize.Min),
+      horizontalArrangement = Arrangement.Center,
+      verticalAlignment = Alignment.Bottom,
+    ) {
+      // user text input. enter sends the message and clears text
+      // shift + enter adds a new line and expands the text field
+      TextArea(
+        value = textValue,
+        onValueChange = { newText -> textValue = newText },
+        modifier =
+          Modifier.weight(1f).heightIn(min = 56.dp).padding(4.dp).onPreviewKeyEvent { event ->
+            when {
+              (event.key == Key.Enter || event.key == Key.NumPadEnter) &&
+                event.type == KeyEventType.KeyDown -> {
+                if (event.isShiftPressed) {
+                  val newText =
+                    textValue.text.replaceRange(
+                      textValue.selection.start,
+                      textValue.selection.end,
+                      "\n",
+                    )
+                  val newSelection = TextRange(textValue.selection.start + 1)
+                  textValue = TextFieldValue(newText, newSelection)
+                  true
+                } else {
+                  sendMessage()
+                  true
+                }
+              }
+              else -> false
+            }
+          },
+        placeholder = { Text("Start your conversation") },
+        keyboardActions = KeyboardActions.Default,
+        maxLines = Int.MAX_VALUE,
```

Comment:
Did you mean to leave these defaults in?

Reply:
I did initially because I was trying different things for the "shift+enter" functionality. I just realized that it doesn't make a difference, so I will remove it. Thanks for pointing it out!